File: tests/fixtures/typescript/store/store.ts
import type { Handler } from "../kinds";

/** One entry in the store. */
export class Entry {
  value: string;

  constructor(value: string) {
    this.value = value;
  }

  describe(): string {
    return this.value.toUpperCase();
  }
}

/** A tiny key-value store, implementing an interface from another file. */
export class Store implements Handler {
  private entries = new Map<string, Entry>();

  /**
   * Read one entry.
   *
   * Two lines of documentation, to prove newlines survive.
   */
  get(key: string): Entry | undefined {
    return this.entries.get(key);
  }

  put(key: string, value: string): void {
    this.entries.set(key, new Entry(value));
  }

  handle(key: string): boolean {
    return this.get(key) !== undefined;
  }

  #evict(): void {
    this.entries.clear();
  }
}

/** A free function that calls a method, so `calls` has an edge to find. */
export function warm(store: Store): Entry | undefined {
  return store.get("warm");
}

File: tests/fixtures/python/store/store.py
"""A class with methods, so that `parent` has something to disagree with the
file about. Unlike Go, the class encloses them: span nesting is enough here.
"""

import re as _re
from typing import Dict


class Entry:
    """One entry in the store."""

    def __init__(self, value: str) -> None:
        self.value = value

    def describe(self) -> str:
        """Render an entry."""
        return _re.sub(r"\s+", " ", self.value.upper())


class Store:
    """A tiny key-value store."""

    def __init__(self) -> None:
        self.entries: Dict[str, Entry] = {}

    def get(self, key: str) -> Entry | None:
        """Read one entry.

        Two lines of documentation, to prove newlines survive.
        """
        return self.entries.get(key)

    def put(self, key: str, value: str) -> None:
        """Write one entry."""
        self.entries[key] = Entry(value)

    def _evict(self) -> None:
        self.entries.clear()


def warm(store: Store) -> Entry | None:
    """A free function that calls a method, so `calls` has an edge to find."""
    return store.get("warm")
